### PR TITLE
Tabs

### PR DESCRIPTION
**Pull request by hedita**

### CODE REVIEW

File: src/Components/TabsItem/TabsItem.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+function TabsItem({ title, content, icon }) {
+  const [clicked, setClicked] = useState(false);
+
+  const handleContentVisibility = () => {
+    setClicked(!clicked);
+  };
+
+  return (
+    <>
```
Comment (takrami) [inline src/Components/TabsItem/TabsItem.jsx:13]: why fragment? :sob:

File: src/Components/TabsItem/TabsItem.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      <div onClick={handleContentVisibility} className="tabs">
+        <i className="tabs-icon">
+          <Icons icon={icon} />
+        </i>
+        <p className="tabs-title">{title}</p>
```
Comment (takrami) [inline src/Components/TabsItem/TabsItem.jsx]: use a proper html tag here.

File: src/Components/TabsItem/TabsItem.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      </div>
+      {clicked ? (
+        <div className="tab-content">
+          <h3>Heading</h3>
```
Comment (takrami) [inline src/Components/TabsItem/TabsItem.jsx]: do you have h1 and h2 in this page?

File: src/Components/TabsItem/TabsItem.scss [hunk 4]
```diff
@@ -0,0 +1,56 @@
+// .tabs-item-container {
```
Comment (takrami) [inline src/Components/TabsItem/TabsItem.scss]: remove commented code.

File: src/Components/TabsItem/TabsItem.scss [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+// }
+
+// .tabs-content {
+//   text-align: center;
+//   line-height: 20px;
+//   margin: 20px;
+// }
+
+
+.tabs {
```
Comment (takrami) [inline src/Components/TabsItem/TabsItem.scss]: please refactor all scss and use scss variable.

File: src/Components/Icons/Icons.jsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCode } from "@fortawesome/free-solid-svg-icons";
```
Comment (takrami) [inline src/Components/Icons/Icons.jsx]: you can import all of them together like:
import 
{faCode, faFolderOpen} from ''"

**Review by takrami — CHANGES_REQUESTED** [review]
Please resolve the comments.

File: src/Components/TabsItem/TabsItem.jsx (hunk 3 above)
Comment (hedita) [reply src/Components/TabsItem/TabsItem.jsx]: I changed it to title.

File: src/Components/TabsItem/TabsItem.jsx (hunk 2 above)
Comment (hedita) [reply src/Components/TabsItem/TabsItem.jsx]: I changed it to title.

File: src/Components/TabsItem/TabsItem.scss (hunk 4 above)
Comment (hedita) [reply src/Components/TabsItem/TabsItem.scss]: Done.

File: src/Components/Icons/Icons.jsx (hunk 6 above)
Comment (hedita) [reply src/Components/Icons/Icons.jsx]: Done.

File: src/Components/TabsItem/TabsItem.scss (hunk 5 above)
Comment (hedita) [reply src/Components/TabsItem/TabsItem.scss]: Done.

File: src/Components/Icons/Icons.jsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCode,faFolderOpen,faEnvelope, faPencilSquare, faBarChart } from "@fortawesome/free-solid-svg-icons";
+import "./Icons.scss";
+
+function Icons({ icon }) {
```
Comment (blacksrc) [inline src/Components/Icons/Icons.jsx]: This function just returns one icon. Fix the naming please.

File: src/Components/TabsItem/TabsItem.jsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+          <Icons icon={icon} />
+        </i>
+        <title className="tabs-title">{title}</title>
+      </div>
+      {clicked ? (
+        <div className="tab-content">
+          <title>Heading</title>
+          <p>{content}</p>
+        </div>
+      ) : null}
```
Comment (blacksrc) [inline src/Components/TabsItem/TabsItem.jsx]: Can be?
```suggestion
      {clicked && (
        <div className="tab-content">
          <title>Heading</title>
          <p>{content}</p>
        </div>
      )}
```

File: src/Components/TabsItem/TabsItem.jsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+function TabsItem({ title, content, icon }) {
+  const [clicked, setClicked] = useState(false);
+
+  const handleContentVisibility = () => {
+    setClicked(!clicked);
+  };
+
+  return (
+    <>
+      <div onClick={handleContentVisibility} className="tabs">
```
Comment (blacksrc) [inline src/Components/TabsItem/TabsItem.jsx]: `tabs` is not proper naming here. It only contains the tab title. Maybe `tab-title` instead?

File: src/Components/TabsItem/TabsItem.jsx [hunk 10]
```diff
@@ -0,0 +1,30 @@
+import React, { useState } from "react";
+import "./TabsItem.scss";
+import Icons from "../Icons/Icons";
+
+function TabsItem({ title, content, icon }) {
+  const [clicked, setClicked] = useState(false);
```
Comment (blacksrc) [inline src/Components/TabsItem/TabsItem.jsx]: Could be a better naming.
```suggestion
  const [isSelected, setIsSelected] = useState(false);
```

File: src/Components/TabsItem/TabsItem.jsx [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+    setClicked(!clicked);
+  };
+
+  return (
+    <>
+      <div onClick={handleContentVisibility} className="tabs">
+        <i className="tabs-icon">
+          <Icons icon={icon} />
+        </i>
+        <title className="tabs-title">{title}</title>
```
Comment (blacksrc) [inline src/Components/TabsItem/TabsItem.jsx]: It's a single tab title.

**Review by blacksrc — CHANGES_REQUESTED** [review]
What are these empty files?
image

File: package.json [hunk 12]
```diff
@@ -4,6 +4,12 @@
   "description": "",
   "main": "index.js",
   "scripts": {
+    "dev": "vite",
+    "build": "vite build",
+    "preview": "vite preview",
+    "lint": "eslint \"src/**/*.{js,jsx}\" --quiet",
+    "format": "prettier --write \"src/**/*.{js,jsx}\"",
+    "sass": "node-sass -w src/scss/ -o src/dist/css --recursive ",
     "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (blacksrc) [inline package.json]: Why you have this script?

File: package.json [hunk 13]
```diff
@@ -4,6 +4,12 @@
   "description": "",
```
Comment (blacksrc) [inline package.json]: Please write a description here.

File: package.json [hunk 14]
```diff
@@ -15,7 +21,11 @@
     "vite": "^4.1.4"
   },
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^6.3.0",
+    "@fortawesome/free-solid-svg-icons": "^6.3.0",
+    "@fortawesome/react-fontawesome": "^0.2.0",
     "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react-dom": "^18.2.0",
+    "sass": "^1.58.3"
```
Comment (blacksrc) [inline package.json]: `sass` should be a devDependency I think.

File: src/Components/Tabs/Tabs.scss [hunk 15]
```diff
@@ -0,0 +1,10 @@
+* {
+    box-sizing: border-box;
+    padding: 0;
+    margin: 0;
+   }
```
Comment (blacksrc) [inline src/Components/Tabs/Tabs.scss]: Fix the indentation please.

File: src/Components/TabsItem/TabsItem.scss [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+    border-top: 4px solid $primary-color;
+
+      .tabs-icon {
+        color: $primary-color;
+      }
+  }
+
+  .tabs-title {
+    padding-bottom: $sizing-small;
+  }
```
Comment (blacksrc) [inline src/Components/TabsItem/TabsItem.scss]: Fix the indentation.

File: src/Components/Icons/Icons.jsx (hunk 7 above)
Comment (hedita) [reply src/Components/Icons/Icons.jsx]: Done.

File: src/Components/TabsItem/TabsItem.jsx (hunk 9 above)
Comment (hedita) [reply src/Components/TabsItem/TabsItem.jsx]: Done.

Comment (hedita) [reply src/Components/TabsItem/TabsItem.jsx]: Done.

File: src/Components/TabsItem/TabsItem.jsx (hunk 11 above)
Comment (hedita) [reply src/Components/TabsItem/TabsItem.jsx]: Done.

File: package.json (hunk 12 above)
Comment (hedita) [reply package.json]: I send a voice.

File: package.json (hunk 14 above)
Comment (hedita) [reply package.json]: I sent a voice.

File: src/Components/Tabs/Tabs.scss (hunk 15 above)
Comment (hedita) [reply src/Components/Tabs/Tabs.scss]: Done.

File: package.json (hunk 13 above)
Comment (hedita) [reply package.json]: I sent a voice.

File: src/Components/TabsItem/TabsItem.scss (hunk 16 above)
Comment (hedita) [reply src/Components/TabsItem/TabsItem.scss]: Done.

File: src/Components/TabsItem/TabsItem.jsx (hunk 3 above)
Comment (hedita) [reply src/Components/TabsItem/TabsItem.jsx]: I changed it.

File: src/Components/TabsItem/TabsItem.jsx (hunk 2 above)
Comment (hedita) [reply src/Components/TabsItem/TabsItem.jsx]: Fixed.

File: package.json (hunk 12 above)
Comment (hedita) [reply package.json]: Done.

File: package.json (hunk 13 above)
Comment (hedita) [reply package.json]: Done.

File: package.json (hunk 14 above)
Comment (hedita) [reply package.json]: Done.

File: src/Components/Icons/Icons.jsx [hunk 17]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCode,faFolderOpen,faEnvelope, faPencilSquare, faBarChart } from "@fortawesome/free-solid-svg-icons";
+import "./Icons.scss";
+
+function Icon({ icon }) {
```
Comment (blacksrc) [inline src/Components/Icons/Icons.jsx]: This component is called `Icon`. The `file`, the `directory` and the `scss` file should be `Icon` as well.

File: src/Components/Icons/Icons.scss [hunk 18]
```diff
@@ -0,0 +1,3 @@
+.tabs-icon {
```
Comment (blacksrc) [inline src/Components/Icons/Icons.scss]: `tab-icon`

File: src/Components/TabsItem/TabsItem.jsx [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+  const [isSelected, setIsSelected] = useState(false);
+
+  const handleContentVisibility = () => {
+    setIsSelected(!isSelected);
+  };
+
+  return (
+    <>
+      <div onClick={handleContentVisibility} className="tab-title">
+        <i className="tabs-icon">
```
Comment (blacksrc) [inline src/Components/TabsItem/TabsItem.jsx]: `tab-icon`

File: src/Components/TabsItem/TabsItem.scss [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+.tab-title {
+  float: left;
+  padding: $sizing-small;
+  padding: $sizing-small $sizing-large $sizing-small $sizing-medium;
+  text-align: center;
+
+  &:hover {
+    border-top: 3px solid $primary-color;
+
+    .tabs-icon {
```
Comment (blacksrc) [inline src/Components/TabsItem/TabsItem.scss]: `tab-icon`

File: src/Components/Icons/Icons.jsx (hunk 17 above)
Comment (hedita) [reply src/Components/Icons/Icons.jsx]: Done.

File: src/Components/Icons/Icons.scss (hunk 18 above)
Comment (hedita) [reply src/Components/Icons/Icons.scss]: Done.

File: src/Components/TabsItem/TabsItem.jsx (hunk 19 above)
Comment (hedita) [reply src/Components/TabsItem/TabsItem.jsx]: Done.

File: src/Components/TabsItem/TabsItem.scss (hunk 20 above)
Comment (hedita) [reply src/Components/TabsItem/TabsItem.scss]: Done.

**Review by blacksrc — APPROVED** [review]

File: src/Components/TabsItem/TabsItem.jsx (hunk 2 above)
Comment (takrami) [reply src/Components/TabsItem/TabsItem.jsx]: this is a title. we shouldn't use p for a title.

File: src/Components/TabsItem/TabsItem.jsx [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+    setIsSelected(!isSelected);
+  };
+
+  return (
+    <>
+      <div onClick={handleContentVisibility} className="tab-title">
+        <i className="tab-icon">
+          <Icon icon={icon} />
+        </i>
+        <h1 className="tab-heading">{title}</h1>
```
Comment (takrami) [inline src/Components/TabsItem/TabsItem.jsx]: i think this is inside a loop, am i right? if yes so it means you have multiple h1 which is wrong. h1 should be just one time in a page.

Comment (hedita) [reply src/Components/TabsItem/TabsItem.jsx]: Done.

**Review by takrami — APPROVED** [review]

**Pull request merged**